### DISCUSSION TITLE
Keycloak does not work

### DISCUSSION THREAD

**Discussion by Dirky14** [Lissy93/dashy#678] [Help and Support]
Hello ! I just tested Dashy and it's a good webapp for what I needed, but I have a little problem. I juste configured Keycloak with the help of the documentation in the wiki, but I don't have any login when I want to access to my site, it just connect to my dashboard and don't ask me for my credentials. Can you help me ? Thanks !

Here is the part of my conf.yml about the keycloak config : [https://pastebin.com/NmR8Nyu7](https://pastebin.com/NmR8Nyu7)

**Comment by lissy93** [comment]
Heya @Dirky14 
That's really weird, since usually once Keycloak is enabled the app won't be able to load at all without a valid token. And if the Keycloak info is incorrect or instance is broken, then Dashy won't load at all. 

Would you mind sharing a bit more info about your setup? Are you running in Docker, if so with the official image? Which version of Dashy and Keycloak are you using?

**Reply by Dirky14** [reply]
Hello, thanks for your response. 

I'm running Dashy on docker, in this container [https://hub.docker.com/r/lissy93/dashy/](https://hub.docker.com/r/lissy93/dashy/) (I think it's the official one ?) and I use the version 16.1.1 of Keycloak (from this docker container : jboss/keycloak:latest), that's why I use the legacySupport : true on my config.

**Reply by itsmejoeeey** [reply]
I saw you mention on #1166 that:
> The login part worked after downgrading from 2.1.1. to 2.1.0

I had the same issue with my Keycloak authentication configuration not applying properly (allowing access without login). I believe it is caused because the latest 2.1.1 docker image does not `yarn build` on startup anymore - potentially indicative of issues with the pipeline.

Downgrading to 2.1.0 worked for me, as did the suggested advice to [change the entrypoint](https://github.com/Lissy93/dashy/issues/1051#issuecomment-1386959639).

See related issue:
#1051: No longer builds on startup
